diff -Naur crazycat-2017-11-13.orig/linux/Makefile crazycat-2017-11-13/linux/Makefile
--- crazycat-2017-11-13.orig/linux/Makefile	2017-12-03 20:47:44.000000000 +0100
+++ crazycat-2017-11-13/linux/Makefile	2017-12-20 09:48:15.516826488 +0100
@@ -117,6 +117,9 @@
 	# Copy videobuf-res module
 	cp -a "$(SRCDIR)/drivers/media/v4l2-core/videobuf-res.c" "drivers/media/v4l2-core/"
 	cp -a "$(SRCDIR)/include/media/videobuf-res.h" "include/media/"
+	mkdir -p  drivers/media/dvb_tv/
+	# Copy dvb_tv driver
+	cp -a $(SRCDIR)/drivers/media/dvb_tv "drivers/media/"
 	-rm -f .patches_applied .linked_dir .git_log.md5
 
 clean:

