diff --git a/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp b/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp
index 31e62ad..7f34805 100644
--- a/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp
+++ b/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp
@@ -1146,6 +1146,8 @@ bool CDVDDemuxFFmpeg::SeekTime(double time, bool backwards, double *startpts)
   if (m_pFormatContext->start_time != (int64_t)AV_NOPTS_VALUE && !ismp3 && !m_bSup)
     seek_pts += m_pFormatContext->start_time;
 
+  CLog::Log(LOGDEBUG, LOGVIDEO, "CDVDDemuxFFmpeg::%s - seek pts:%0.3f backwards:%d start_time:%0.3f",
+             __FUNCTION__, (double)seek_pts / AV_TIME_BASE, backwards, (double)(m_pFormatContext->start_time) / AV_TIME_BASE);
   int ret;
   {
     CSingleLock lock(m_critSection);
@@ -1177,7 +1179,7 @@ bool CDVDDemuxFFmpeg::SeekTime(double time, bool backwards, double *startpts)
   if(m_currentPts == DVD_NOPTS_VALUE)
     CLog::Log(LOGDEBUG, "%s - unknown position after seek", __FUNCTION__);
   else
-    CLog::Log(LOGDEBUG, "%s - seek ended up on time %d", __FUNCTION__, (int)(m_currentPts / DVD_TIME_BASE * 1000));
+    CLog::Log(LOGDEBUG, "%s - seek ended up on time %0.3f", __FUNCTION__, m_currentPts / DVD_TIME_BASE);
 
   // in this case the start time is requested time
   if (startpts)
@@ -1772,6 +1774,7 @@ int64_t CDVDDemuxFFmpeg::GetChapterPos(int chapterIdx)
 
 bool CDVDDemuxFFmpeg::SeekChapter(int chapter, double* startpts)
 {
+  CLog::Log(LOGDEBUG, "CDVDDemuxFFmpeg::%s - Seeking chapter:%d startpts:%p ", __FUNCTION__, chapter, startpts);
   if(chapter < 1)
     chapter = 1;
 
@@ -1795,11 +1798,25 @@ bool CDVDDemuxFFmpeg::SeekChapter(int chapter, double* startpts)
     return false;
 
   if(chapter < 1 || chapter > (int)m_pFormatContext->nb_chapters)
+{  CLog::Log(LOGDEBUG, "CDVDDemuxFFmpeg::%s - wrong chapter:%d max:%d", __FUNCTION__, chapter, (int)m_pFormatContext->nb_chapters);
     return false;
-
+}
   AVChapter *ch = m_pFormatContext->chapters[chapter-1];
   double dts = ConvertTimestamp(ch->start, ch->time_base.den, ch->time_base.num);
-  return SeekTime(DVD_TIME_TO_MSEC(dts), true, startpts);
+  bool rtn = SeekTime(DVD_TIME_TO_MSEC(dts), true, startpts);
+  if (rtn)
+  {
+    double startpts_b;
+    if (SeekTime(DVD_TIME_TO_MSEC(dts), false, &startpts_b))
+    {
+      if (*startpts - dts < dts -startpts_b)
+        rtn = SeekTime(DVD_TIME_TO_MSEC(dts), true, startpts);
+      else
+        *startpts = startpts_b;
+    }
+  }
+  CLog::Log(LOGDEBUG, "CDVDDemuxFFmpeg::%s - seeking chapter:%d start:%0.3f key-frame:%0.3f ", __FUNCTION__, chapter, DVD_TIME_TO_MSEC(dts) / 1000.0, m_currentPts / DVD_TIME_BASE);
+  return rtn;
 }
 
 std::string CDVDDemuxFFmpeg::GetStreamCodecName(int iStreamId)
